### ISSUE TITLE
[USABILITY] My Events: Add Events you administer to the view

### ISSUE THREAD

**Issue by vburgett** [rokwire/illinois-app#4082]
When a user goes to Browse > My Events, they see events they have starred. 

Please add "Starred" _and_ "Admin" to the "event type" filters used for My Events.

**Comment by mihail-varbanov** [comment]
I think you did not express yourself correctly. If I _**add "Starred" and "Admin" to the "event type" filters used for My Events**_, My Events panel will display the events that are both starred and administered by the user at the same time. I can do this quickly and easily but I suspect you did not mean this. What you probably want to happen is to **add "Starred" OR "Admin" to the "event type" filters used for My Events**, i.e. My Events panel displays events that are either starred or being administered by the user.

Тhis is currently not at all easily achievable because:

1. The backend does not support such logic at the moment. All filters are assumed to be applied with AND clause, i.e. the events in the response would match ALL individual filters applied. Implementing a logic to match ANY individual filter would lead to designing and  implementation of arbitrary boolean expressions for event filters. Examples of such arbitrary boolean expressions are:
```
Event is: {starred} OR {admin}
Event is: {free} AND ({in-person} OR {hybrid}))
Event is: ({free} AND ({in-person} OR {hybrid}))) OR ({public} AND {admin})
```
2. The current app's UI (i.e. the event attributes panel) is also not ready to support such boolean expressions. Using lists if checkboxes imply the use of AND clauses for individual filter entries. Even if we do not implement definitions of arbitrary boolean expressions, we need to provide a way to apply AND or OR clause for filter entries, that would complicate the interface significantly. The example for a similar UI I can think of right now is rules definition in the mail client applications.

So, to summarize, in my opinion this is a hard and long-term task that should be planned early enough before a release.

**Comment by vburgett** [comment]
Thanks for the explanation, @mihail-varbanov.

To solve this issue for now, could events created via the admin website automatically be favorited in the UIUC app for those listed in the "Event Admin NetIDs" field? The users could choose to unstar their event in the UIUC app.  

If this is an option, do I submit a new ticket via this repository or the Illinois admin one?

![Screenshot 2024-06-11 at 1 17 31 PM](https://github.com/rokwire/illinois-app/assets/1044905/d958d86e-398b-4402-80c2-21bf82a14a9f)

**Comment by mihail-varbanov** [comment]
Hi @vburgett. Yes, you should rather create an issue for Illinois Admin app. If feasible at all, most likely it would reflect to creating an issue for the backend as I do not believe there is suitable API that would accomplish that. I created the Admin issue for you: [#634](https://github.com/rokwire/illinois_admin/issues/634).

**Comment by dobromirdobrev** [comment]
Hi @vburgett ,

We closed the [Admin app issue #634](https://github.com/rokwire/illinois_admin/issues/634) because the functionality for creating an events will be transferred to the UIUC app.

Also, currently in the UIUC app when an event is successfully created, we show a prompt asking the user if one wants to favourite the event:

<img width="359" alt="Image" src="https://github.com/user-attachments/assets/79e17526-ea61-432e-bcbf-abc183f93dba" />

So, I would say that we have this functionality in the current UI of the UIUC app.

Having in mind the above, I see three options here:

1. The prompt to stay as is - no changes
2. Remove the prompt and "automatically" star the event
3. Wait for the new UI related to creating events in which we can point more than one admin (more than one NetID) and star the event for all the admins.

Please, let us know how we should proceed with this. Thanks! cc @pmarkhennessy

**Comment by vburgett** [comment]
For now, let's go with options 1 and 3. Thank you, @dobromirdobrev.
